### PR TITLE
ARROW-12701: [Website][Release] Include Rust contributors, committers, and commits in release notes

### PR DESCRIPTION
**Pull request by thisisnic**
Note: this PR does not include DataFusion commits because it is not yet clear what release schedule and tagging conventions will be used for the DataFusion repo.

### CODE REVIEW

File: dev/release/post-03-website.sh
Comment (jonkeane) [inline dev/release/post-03-website.sh:60]: Should this have the ARROW_DF_DIR in it as well?

File: dev/release/post-03-website.sh
Comment (jonkeane) [inline dev/release/post-03-website.sh:72]: This is some `awk` right here!

File: dev/release/post-03-website.sh
Comment (jonkeane) [inline dev/release/post-03-website.sh:85]: Would it make sense to pull this `awk` command out into a variable so we only have to specify it once?

File: dev/release/post-03-website.sh
Comment (jonkeane) [inline dev/release/post-03-website.sh:118]: Do you know why `${contributors_command_line}` was called here but then `${contributors}` was used below? I tried to work out a reasoning why but couldn't quite grok what was going on here.

**Review by jonkeane — COMMENTED** [review]
Looks good, a few comments

File: dev/release/post-03-website.sh
Comment (thisisnic) [reply dev/release/post-03-website.sh:118]: Originally there were two variables:

1. the code to run to get the list of commuters and number of commits
2. the output of running the code from 1

I think it was done like that before so that the variable 1 could be displayed in the output.  However, now we use longer commands to generate the list of committers/commits, it's no longer desirable to display it and so this has been simplified by just using a single variable.

File: dev/release/post-03-website.sh
Comment (thisisnic) [reply dev/release/post-03-website.sh:85]: Good point, I will do this!

File: dev/release/post-03-website.sh
Comment (thisisnic) [reply dev/release/post-03-website.sh:60]: Yep - I need to get in touch with the DF folks about their tagging conventions before I can do this.

Comment (jonkeane) [reply dev/release/post-03-website.sh:60]: Ok cool, maybe a comment in here explaining that would be good to make it extra clear it's not an omission?

File: dev/release/post-03-website.sh
Comment (jonkeane) [reply dev/release/post-03-website.sh:118]: Ok, cool makes sense thanks!

File: dev/release/post-03-website.sh
Comment (thisisnic) [reply dev/release/post-03-website.sh:72]: Thank goodness for SO!

File: dev/release/post-03-website.sh
Comment (thisisnic) [reply dev/release/post-03-website.sh:60]: Absolutely.

Comment (thisisnic) [reply dev/release/post-03-website.sh:61]: I've added a comment and opened up a ticket here to clarify with the DataFusion folks: https://github.com/apache/arrow-datafusion/issues/770

File: dev/release/post-03-website.sh
Comment (thisisnic) [reply dev/release/post-03-website.sh]: Done

**Comment by ianmcook** [comment]
Thanks for doing this @thisisnic! Could you also please change how `n_commits` is calculated? It should iterate over `directories` and total up all the commits.

File: dev/release/post-03-website.sh
Comment (ianmcook) [inline dev/release/post-03-website.sh:60]: Add a comment here to provide context:
```suggestion
# Previously this code counted commits, committers, and contributors only to the
# apache/arrow repository. After the Rust implementation of Arrow was moved to
# the separate apache/arrow-rs repository, this code was modified to also count
# commits, committers, and contributors to that repository.
#
# TODO: Add DataFusion directory and git ranges here once git tagging convention
# for apache/arrow-datafusion repo is confirmed with DF team
#
# TODO: Consider counting other apache/arrow-* repos here as tagging conventions
# permit (arrow-cookbook, arrow-site)

```

File: dev/release/post-03-website.sh
Comment (ianmcook) [inline dev/release/post-03-website.sh:153]:
```suggestion
The following Apache committers merged contributed patches to Arrow repositories.
```

**Comment by thisisnic** [comment]
> Thanks for doing this @thisisnic! Could you also please change how `n_commits` is calculated? It should iterate over `directories` and total up all the commits.

How do you mean, sorry?  I thought that's what it currently does.

File: dev/release/post-03-website.sh
Comment (ianmcook) [inline dev/release/post-03-website.sh:151]:
```suggestion
This release includes ${n_commits} commits from ${n_contributors} distinct contributors in ${#directories[@]} repositories.
```

**Comment by thisisnic** [comment]
> Thanks for doing this @thisisnic! Could you also please change how `n_commits` is calculated? It should iterate over `directories` and total up all the commits.

Done

File: dev/release/post-03-website.sh
Comment (thisisnic) [inline dev/release/post-03-website.sh:27]:
```suggestion
# TODO: add this back in when we have datafusion releases included
#ARROW_DF_DIR="${ARROW_DIR}/../arrow-datafusion"
```

File: dev/release/post-03-website.sh
Comment (ianmcook) [reply dev/release/post-03-website.sh:72]: Per Nic, It seems like the `arrow-datafusion` releases will not necessarily be in sync with the `arrow` and `arrow-rs` releases, so we should leave DataFusion out of this for now and revisit later as noted in the `TODO` comments.

**Review by ianmcook — APPROVED** [review]

**Review by kszucs — APPROVED** [review]
Thanks @thisisnic!

**Comment by kszucs** [comment]
Created a follow-up ticket to port the release note generation to archery at some point: https://issues.apache.org/jira/browse/ARROW-13483

**Comment by kszucs** [comment]
Regenerated the release not for 5.0.0 with this patch: https://github.com/apache/arrow-site/pull/131

**Comment by ianmcook** [comment]
> Regenerated the release note for 5.0.0 with this patch: [apache/arrow-site#131](https://github.com/apache/arrow-site/pull/131)

Thanks! @thisisnic there's a problem with the Rust committers; please see https://github.com/apache/arrow-site/pull/131/files#r678709259

**Comment by ianmcook** [comment]
@thisisnic I manually fixed the problem with the Rust committers and applied the fix in https://github.com/apache/arrow-site/pull/133. So we're OK for the 5.0.0 release and there's no great urgency, but let's fix the problem before we merge this PR.

**Comment by thisisnic** [comment]
@ianmcook Do you remember what you did to get the correct list on this ticket? I've been looking into it but a lot of what I find online is people complaining that they can't find that information after a squash & merge.

**Comment by thisisnic** [comment]
Closing this ancient PR and starting again in a new one

**Pull request closed without merging**
